src/System.Collections.Specialized/tests/HybridDictionary/HybridDictionary.KeysTests.cs

```diff
@@ -56,12 +61,51 @@ public override void ICollection_NonGeneric_CopyTo_NonZeroLowerBound(int count)
             // the lower bounds of the destination array for count > 10
             if (count < 10)
             {
-                Assert.Throws<ArgumentException>("array", () => collection.CopyTo(arr, 0));
+                Assert.Throws(Helpers.IsDesktopJob ? typeof(IndexOutOfRangeException) : typeof(ArgumentException), () => collection.CopyTo(arr, 0));
             }
             else
             {
                 Assert.Throws<IndexOutOfRangeException>(() => collection.CopyTo(arr, 0));
             }
         }
+
+        [Theory]
+        [MemberData(nameof(ValidCollectionSizes))]
+        public override void ICollection_NonGeneric_CopyTo_IndexEqualToArrayCount_ThrowsArgumentException(int count)
+        {
+            ICollection collection = NonGenericICollectionFactory(count);
+            object[] array = new object[count];
+            if (count > 0)
+                Assert.Throws(Helpers.IsDesktopJob && count < 10 ? typeof(IndexOutOfRangeException) : typeof(ArgumentException), () => collection.CopyTo(array, count));
```

Comment:
>IsDesktopJob [](start = 38, length = 12)

Rather than magic number 10 is it better to use an expression on collection.Count?  
Although in this case maybe no expression is needed since the CopyTo will always fail.